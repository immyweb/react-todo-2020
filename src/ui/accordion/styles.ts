import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
