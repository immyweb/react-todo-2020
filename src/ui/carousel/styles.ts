import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
