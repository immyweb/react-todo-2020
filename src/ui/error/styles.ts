import glamorous from 'glamorous';

import { H3 } from '../typography/typography';
import { fonts } from '../../styles/fonts';

interface IGProps {
  backgroundColour?: string;
  marginBottom?: number;
}

export const ErrorWrapper = glamorous.div<IGProps>(
  {
    label: 'error-wrapper',
    position: 'relative',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 200,
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const Heading = glamorous(H3)<IGProps>(
  {
    fontFamily: fonts.$theSunHeavyCondensed,
    textAlign: 'center',
    fontSize: 22,
  },
  ({ marginBottom }) => ({
    marginBottom: marginBottom ? marginBottom : 5,
  }),
);

export const Graphic = glamorous.div({
  label: 'graphic',
  position: 'relative',
  margin: '0 auto 10px auto',
});
