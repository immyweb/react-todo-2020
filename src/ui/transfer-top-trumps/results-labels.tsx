import * as React from 'react';

import { ResultsProfileLabel, ResultsProfileLabelContainer } from './styles';

import { IPlayer } from './types';

export const ResultsLabels: React.SFC<{ sortedProfiles: IPlayer[] }> = ({ sortedProfiles }) => {
  const profiles = sortedProfiles.map((profile, i) => {
    if (profile !== undefined) {
      const { fname, sName } = profile;
      return <ResultsProfileLabel key={i}>{`${fname} ${sName}`}</ResultsProfileLabel>;
    }
    return <div key={i} />;
  });
  return <ResultsProfileLabelContainer>{profiles}</ResultsProfileLabelContainer>;
};
