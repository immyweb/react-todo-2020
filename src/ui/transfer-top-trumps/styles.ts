import glamorous, { Span, Div } from 'glamorous';
import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { H3, H2, P } from '../typography/typography';
import Button from '../buttons/buttons';
import { mediaQueries } from '../../styles/breakpoints';

// Common Elements

export const TTContainer = glamorous.section({
  label: 'ttt-container',
  background: `linear-gradient(to bottom, ${colours.$black} 0%, ${colours.$offBlack} 20%, ${
    colours.$offBlack
  } 50%, ${colours.$black} 81%, ${colours.$black} 100%)`,
  width: '100%',
  position: 'relative',
  marginBottom: 15,
  [mediaQueries.tabletPortrait]: {
    minHeight: 'auto',
  },
});

export const TTTopGradient = glamorous.div({
  label: 'ttt-top-gradient',
  width: '100%',
  height: 84,
  background: `linear-gradient(to bottom, ${colours.$offBlack}, transparent)`,
  position: 'absolute',
  top: 0,
  zIndex: 2,
});

export const TTBottomGradient = glamorous.div({
  label: 'ttt-top-gradient',
  width: '100%',
  height: 130,
  background: `linear-gradient(rgba(34, 37, 38, 0.9), rgba(34, 37, 38, 0.9))`,
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
});

export const Inner = glamorous(Div)({
  label: 'ttt-inner',
  margin: '0 45px',
  padding: '20px 0',
  position: 'relative',
  zIndex: 3,
});

// Headline

export const TTHeadline = glamorous(H3)({
  label: 'ttt-card-first-name',
  fontSize: 18,
  textTransform: 'uppercase',
  textAlign: 'center',
  marginTop: 0,
  fontFamily: fonts.$theSunHeavyCondensed,
  [mediaQueries.tabletPortrait]: {
    fontSize: 18,
  },
});

export const TTSubHeadline = glamorous(H2)({
  label: 'ttt-card-surname',
  fontSize: 48,
  textTransform: 'uppercase',
  textAlign: 'center',
  [mediaQueries.tabletPortrait]: {
    fontSize: 46,
  },
});

// Buttons

export const TTRoundBtn = glamorous(Button)<{
  size?: number;
  bkgndTop: string;
  bkgndBottom: string;
  borderWidth?: number;
}>(
  {
    label: 'ttt-round-button',
    padding: 0,
    borderRadius: '50%',
    margin: '0 15px',
    boxShadow: '0 0 2px 0 rgba(27, 27, 27, 0.5)',
    pointerEvents: 'auto',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  ({ size, bkgndTop, bkgndBottom, borderWidth }) => ({
    width: size ? size : 76,
    height: size ? size : 76,
    border: borderWidth
      ? `${borderWidth}px solid ${colours.$white}`
      : `3px solid ${colours.$white}`,
    backgroundImage: `linear-gradient(to bottom, ${bkgndTop}, ${bkgndBottom})`,
    ':hover:enabled': {
      backgroundImage: `linear-gradient(to bottom, ${bkgndBottom}, ${bkgndTop})`,
    },
    ':hover:disabled': {
      backgroundImage: `linear-gradient(to bottom, ${bkgndTop}, ${bkgndBottom})`,
      cursor: 'default',
    },
  }),
);

export const TTLozengeBtn = glamorous(Button)<{
  bkgndTop: string;
  bkgndBottom: string;
}>(
  {
    label: 'ttt-lozenge-button',
    padding: 0,
    height: 60,
    width: '100%',
    borderRadius: 30,
    border: `2px solid ${colours.$white}`,
    textTransform: 'uppercase',
    fontSize: 22,
    display: 'flex',
    justifyContent: 'center',
    [mediaQueries.tabletPortrait]: {
      fontSize: 18,
      height: 40,
    },
  },
  ({ bkgndTop, bkgndBottom }) => ({
    backgroundImage: `linear-gradient(to bottom, ${bkgndTop}, ${bkgndBottom})`,
    ':hover:enabled': {
      backgroundImage: `linear-gradient(to bottom, ${bkgndBottom}, ${bkgndTop})`,
    },
  }),
);

export const TTLozengeBtnIcon = glamorous.img({
  label: 'ttt-lozenge-button-icon',
  display: 'inline-block',
  width: 30,
  height: 30,
  marginRight: 10,
  [mediaQueries.tabletPortrait]: {
    width: 20,
    height: 20,
  },
});

export const BtnHolder = glamorous.div({
  label: 'ttt-btn-holder',
  [mediaQueries.tabletPortrait]: {
    width: 200,
    margin: '0 auto',
  },
});

export const Divider = glamorous.div<{
  height?: number;
  offsetTop?: number;
  mobileHide?: boolean;
}>(
  {
    label: 'ttt-divider',
    width: 1,
    opacity: 0.5,
    background: colours.$white,
  },
  ({ height, offsetTop, mobileHide }) => ({
    height: height ? height : 71,
    marginTop: offsetTop ? offsetTop : 0,
    display: mobileHide ? 'none' : 'inline-block',
    [mediaQueries.tabletPortrait]: {
      display: 'inline-block',
    },
  }),
);

// Gradient

export const GradientContainer = glamorous.div<{
  size?: number;
  top?: number;
  opacity?: number;
}>(
  {
    label: 'ttt-gradient-container',
    position: 'absolute',
    zIndex: 0,
    width: '100%',
    left: '50%',
    transform: `translateX(-50%)`,
    [mediaQueries.tabletPortrait]: {
      width: 300,
    },
  },
  ({ size, top, opacity }) => ({
    width: size ? size : '100%',
    top: top ? top : 10,
    opacity: opacity ? opacity : 1.0,
  }),
);

export const GradientGroup = glamorous.div<{ width: string }>(
  {
    label: 'ttt-gradient-group',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    margin: '0 auto',
  },
  ({ width }) => ({
    width,
  }),
);

export const GradientLine = glamorous.div<{
  direction: string;
  teamColour: string;
  height?: number;
}>(
  {
    label: 'ttt-gradient',
    width: 145,
    marginBottom: 3,
  },
  ({ direction, teamColour, height }) => ({
    backgroundImage: `linear-gradient(to ${direction}, rgba(32, 32, 32, 0), ${teamColour})`,
    height: height ? height : 8,
  }),
);

export const BadgeImageHolder = glamorous.div({
  label: 'ttt-card-badge-image-holder',
  width: '100%',
  textAlign: 'center',
  position: 'relative',
  minHeight: 50,
});

export const Badge = glamorous.div<{ size?: number }>(
  {
    label: 'ttt-card-badge',
    margin: '0 auto',
    position: 'relative',
    top: 0,
    minHeight: 50,
  },
  ({ size }) => ({
    width: size ? size : 50,
  }),
);

export const BadgeImage = glamorous.img({
  label: 'ttt-card-badge-image',
  display: 'inline-block',
});

// Intro Screen

export const IntroHeader = glamorous(H3)({
  label: 'ttt-intro-header',
  color: colours.$white,
  textTransform: 'uppercase',
  fontFamily: fonts.$theSunMedium,
  fontSize: 30,
  textAlign: 'center',
  marginBottom: 7,
  [mediaQueries.tabletPortrait]: {
    fontSize: 22,
  },
});

export const IntroContent = glamorous(P)({
  label: 'ttt-intro-content',
  textTransform: 'uppercase',
  textAlign: 'center',
  fontSize: 18,
  lineHeight: 1.11,
  fontFamily: fonts.$theSunMedium,
  [mediaQueries.tabletPortrait]: {
    width: 250,
    margin: '0 auto',
    fontSize: 14,
    lineHeight: 1.21,
  },
});

export const IntroInstructions = glamorous.div({
  label: 'ttt-intro-instructions',
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-evenly',
  margin: '25px 0',
  [mediaQueries.tabletPortrait]: {
    justifyContent: 'center',
  },
});

export const IntroInstructionsContent = glamorous.div<{ reverse?: boolean }>(
  {
    label: 'ttt-intro-instructions-content',
    [mediaQueries.tabletPortrait]: {
      display: 'flex',
    },
  },
  ({ reverse }) => ({
    [mediaQueries.tabletPortrait]: {
      flexDirection: reverse ? 'row-reverse' : 'row',
    },
  }),
);

export const IntroInstructionsContentP = glamorous(P)({
  label: 'ttt-intro-instructions-content-p',
  color: colours.$white,
  textTransform: 'uppercase',
  textAlign: 'center',
  marginTop: 10,
  fontSize: 14,
  width: 94,
  [mediaQueries.tabletPortrait]: {
    fontSize: 14,
    width: 'auto',
    marginTop: 33,
  },
});

// Card

export const Game = glamorous.section({
  label: 'ttt-game',
});

export const CardContainer = glamorous.div<{ voted?: boolean }>(
  {
    label: 'ttt-card',
    width: '100%',
    position: 'relative',
    paddingTop: 14,
    cursor: 'pointer',
    zIndex: 3,
    minHeight: 485,
    background: colours.$black,
    [mediaQueries.mobileInner]: {
      minHeight: 352,
    },
  },
  ({ voted }) => ({
    display: voted ? 'none' : 'block',
  }),
);

export const CardProfileImage = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'ttt-card-profile-image',
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: 497,
    backgroundBlendMode: 'multiply',
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
    [mediaQueries.mobileInner]: {
      height: 364,
    },
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile}), linear-gradient(to bottom, ${
      colours.$offBlack
    } 0%, transparent 20%, transparent 50%, ${colours.$offBlack} 81%, ${colours.$offBlack} 100%)`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop}), linear-gradient(to bottom, ${
        colours.$offBlack
      } 0%, transparent 20%, transparent 50%, ${colours.$offBlack} 81%, ${colours.$offBlack} 100%)`,
    },
  }),
);

export const CardHeader = glamorous.div({
  label: 'ttt-card-header',
  position: 'relative',
  zIndex: 2,
  minHeight: 75,
});

export const CardFirstname = glamorous(H3)({
  label: 'ttt-card-first-name',
  fontSize: 18,
  textTransform: 'uppercase',
  textAlign: 'center',
  fontFamily: fonts.$theSunHeavyCondensed,
  [mediaQueries.tabletPortrait]: {
    fontSize: 18,
  },
});

export const CardSurname = glamorous(H2)({
  label: 'ttt-card-surname',
  fontSize: 48,
  textTransform: 'uppercase',
  textAlign: 'center',
  lineHeight: 0.96,
  [mediaQueries.tabletPortrait]: {
    fontSize: 48,
    lineHeight: 0.96,
  },
});

export const CardInfo = glamorous.div({
  label: 'ttt-card-info',
  width: '100%',
  padding: 10,
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
});

export const CardLabel = glamorous(Span)({
  label: 'ttt-card-label',
  textTransform: 'uppercase',
  fontFamily: fonts.$theSunRegular,
  textAlign: 'center',
  display: 'block',
  fontSize: 14,
  lineHeight: 1,
});

export const CardContent = glamorous(Span)({
  label: 'ttt-card-content',
  color: colours.$white,
  textTransform: 'uppercase',
  fontFamily: fonts.$theSunMedium,
  display: 'block',
  fontSize: 30,
  lineHeight: '30px',
  textAlign: 'center',
});

export const CardStats = glamorous.div({
  label: 'ttt-card-stats',
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'space-evenly',
  marginTop: 13,
  [mediaQueries.tabletPortrait]: {
    width: 'auto',
    marginTop: 10,
  },
});

export const CardClub = glamorous.div({
  label: 'ttt-card-club',
  display: 'inline-block',
  width: '100%',
  marginBottom: 15,
  [mediaQueries.tabletPortrait]: {
    width: 'auto',
    marginBottom: 0,
  },
});

export const CardPlayer = glamorous.div({
  label: 'ttt-card-player',
  display: 'inline-block',
  padding: '0 20px',
  flex: 1,
  [mediaQueries.tabletPortrait]: {
    flex: 'none',
    padding: 0,
  },
});

// Controls

export const ControlsHolder = glamorous.div({
  label: 'ttt-nav-holder',
  position: 'absolute',
  top: '36%',
  zIndex: 5,
  display: 'flex',
  justifyContent: 'space-between',
  width: '100%',
  pointerEvents: 'none',
  [mediaQueries.tabletPortrait]: {
    justifyContent: 'space-between',
  },
});

// Progress Bar

export const ProgressBarContainer = glamorous.div<{ bkgnd: string }>(
  {
    label: 'ttt-progress-bar',
    width: '100%',
    padding: '15px 0',
    textAlign: 'center',
    borderTop: `1px solid ${colours.$white}`,
  },
  ({ bkgnd }) => ({
    background: bkgnd,
  }),
);

export const ProgressBarInner = glamorous.div({
  label: 'ttt-progress-bar-inner',
  display: 'inline-block',
  position: 'relative',
});

export const ProgressBarIcon = glamorous.div({
  label: 'ttt-progress-bar-icon',
  position: 'absolute',
  bottom: 0,
  right: -10,
});

export const ProgressBarImage = glamorous.div<{
  voted?: boolean;
  active?: boolean;
  selected?: boolean;
}>(
  {
    label: 'ttt-progress-bar-image',
    width: 58,
    height: 58,
    borderRadius: '50%',
    overflow: 'hidden',
    background: colours.$offBlack,
    display: 'inline-block',
    margin: '0 3px',
    [mediaQueries.tabletPortrait]: {
      margin: '0 7px',
    },
  },
  ({ voted, active }) => ({
    opacity: voted || active ? 1.0 : 0.3,
    border: active ? `4px solid ${colours.$white}` : `2px solid ${colours.$white}`,
  }),
);

// Results

export const ResultsInner = glamorous.div({
  label: 'ttt-results-inner',
  margin: '0 15px',
  padding: '20px 0 0',
  position: 'relative',
  zIndex: 3,
  textAlign: 'center',
});

export const Ribbon = glamorous(H3)({
  label: 'ttt-ribbon',
  position: 'relative',
  color: colours.$white,
  textTransform: 'uppercase',
  margin: '0 auto 18px',
  padding: '10px 25px',
  textAlign: 'center',
  backgroundColor: colours.$mainSecondary,
  display: 'inline-block',
  fontSize: 20,
  fontFamily: fonts.$theSunHeavyCondensed,
  [mediaQueries.tabletPortrait]: {
    display: 'none',
  },
  '&::before, &::after': {
    content: '""',
    position: 'absolute',
    display: 'block',
    border: `17px solid ${colours.$mainSecondary}`,
    opacity: 0.8,
    top: 4,
  },
  '&::before': {
    left: '-28px',
    borderRightWidth: '17px',
    borderLeftColor: 'transparent',
  },
  '&::after': {
    right: '-28px',
    borderLeftWidth: '17px',
    borderRightColor: 'transparent',
  },
});

// Chart

export const BarWrapper = glamorous.div({
  label: 'bar-wrapper',
  position: 'relative',
  maxWidth: 320,
  margin: '0 auto',
  [mediaQueries.tabletPortrait]: {
    maxWidth: 'none',
    margin: '0 50px',
  },
});

export const BarContainer = glamorous.div({
  label: 'bar-container',
  position: 'relative',
  width: '100%',
  height: '100%',
});

export const ResultsCharts = glamorous.div({
  label: 'ttt-results-charts',
  position: 'relative',
});

export const ResultsProfileImageContainer = glamorous.div({
  label: 'ttt-results-profile-image-container',
  display: 'flex',
  flexDirection: 'column',
  position: 'absolute',
  left: 0,
  zIndex: 1,
  [mediaQueries.tabletPortrait]: {
    left: 40,
  },
});

export const ResultsProfileImage = glamorous.div({
  label: 'ttt-results-profile-image',
  width: 54,
  height: 54,
  borderRadius: '50%',
  overflow: 'hidden',
  background: colours.$offBlack,
  display: 'inline-block',
  margin: '6px 0',
  border: `2px solid ${colours.$white}`,
});

export const ResultsProfileInner = glamorous.div({
  label: 'ttt-results-profile-inner',
  position: 'relative',
});

export const ResultsProfileIcon = glamorous.div({
  label: 'ttt-results-profile-icon',
  position: 'absolute',
  bottom: 6,
  right: -15,
});

export const ResultsProfileLabelContainer = glamorous.div({
  label: 'ttt-results-profile-label-container',
  display: 'flex',
  flexDirection: 'column',
  position: 'absolute',
  left: 63,
  zIndex: 1,
  [mediaQueries.tabletPortrait]: {
    left: 110,
  },
});

export const ResultsProfileLabel = glamorous.div({
  label: 'ttt-results-profile-label',
  color: colours.$white,
  textTransform: 'uppercase',
  fontFamily: fonts.$theSunHeavyNarrow,
  fontSize: 16,
  textAlign: 'left',
  margin: '6px 0 42px',
});
