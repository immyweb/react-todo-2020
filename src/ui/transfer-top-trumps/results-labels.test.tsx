import * as React from 'react';
import { shallow } from 'enzyme';

import { ResultsLabels } from './results-labels';
import { ResultsProfileLabel } from './styles';

const players = [
  {
    id: 'willian',
    fname: '',
    sName: 'Willian',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg',
    club: 'Chelsea',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png',
    age: 29,
    fee: '£60m',
    position: 'CAM',
    voted: true,
    selected: false,
    active: false,
    idx: 0,
    voteCount: 20,
  },
  {
    id: 'a_sandro',
    fname: 'Alex',
    sName: 'Sandro',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg',
    club: 'Juventus',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png',
    age: 27,
    fee: '£60m',
    position: 'LB',
    voted: true,
    selected: true,
    active: false,
    idx: 1,
    voteCount: 50,
  },
  {
    id: 's_m_savic',
    fname: 'Sergej',
    sName: 'Milinković-Savić',
    imageMobile:
      'https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg',
    imageDesktop:
      'https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg',
    club: 'SS Lazio',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png',
    age: 23,
    fee: '£70m',
    position: 'CM',
    voted: false,
    selected: false,
    active: true,
    idx: 2,
    voteCount: 60,
  },
];

function shallowRender(props?: any) {
  return shallow(<ResultsLabels sortedProfiles={players} {...props} />);
}

describe('<ResultsLabels />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct number of labels', () => {
    const wrapper = shallowRender();
    const labels = wrapper.find(ResultsProfileLabel);
    expect(labels.length).toEqual(3);
    const wrapper2 = shallowRender({ sortedProfiles: [] });
    const labels2 = wrapper2.find(ResultsProfileLabel);
    expect(labels2.length).toEqual(0);
  });

  it('renders empty div if no profile provided', () => {
    const wrapper = shallowRender({ sortedProfiles: [] });
    const labels = wrapper.find(ResultsProfileLabel);
    expect(labels.length).toEqual(0);
  });
});
