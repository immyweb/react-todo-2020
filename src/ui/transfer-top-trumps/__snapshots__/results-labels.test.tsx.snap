// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsLabels /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    sortedProfiles={
      Array [
        Object {
          "active": false,
          "age": 29,
          "club": "Chelsea",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png",
          "fee": "£60m",
          "fname": "",
          "id": "willian",
          "idx": 0,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
          "position": "CAM",
          "sName": "Willian",
          "selected": false,
          "voteCount": 20,
          "voted": true,
        },
        Object {
          "active": false,
          "age": 27,
          "club": "Juventus",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png",
          "fee": "£60m",
          "fname": "Alex",
          "id": "a_sandro",
          "idx": 1,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg",
          "position": "LB",
          "sName": "Sandro",
          "selected": true,
          "voteCount": 50,
          "voted": true,
        },
        Object {
          "active": true,
          "age": 23,
          "club": "SS Lazio",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png",
          "fee": "£70m",
          "fname": "Sergej",
          "id": "s_m_savic",
          "idx": 2,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg",
          "position": "CM",
          "sName": "Milinković-Savić",
          "selected": false,
          "voteCount": 60,
          "voted": false,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)>
           Willian
        </glamorous(div)>,
        <glamorous(div)>
          Alex Sandro
        </glamorous(div)>,
        <glamorous(div)>
          Sergej Milinković-Savić
        </glamorous(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": " Willian",
        },
        "ref": null,
        "rendered": " Willian",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": "Alex Sandro",
        },
        "ref": null,
        "rendered": "Alex Sandro",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": "Sergej Milinković-Savić",
        },
        "ref": null,
        "rendered": "Sergej Milinković-Savić",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)>
             Willian
          </glamorous(div)>,
          <glamorous(div)>
            Alex Sandro
          </glamorous(div)>,
          <glamorous(div)>
            Sergej Milinković-Savić
          </glamorous(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": " Willian",
          },
          "ref": null,
          "rendered": " Willian",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": "Alex Sandro",
          },
          "ref": null,
          "rendered": "Alex Sandro",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": "Sergej Milinković-Savić",
          },
          "ref": null,
          "rendered": "Sergej Milinković-Savić",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
